command/var_lock.go

```diff
@@ -0,0 +1,329 @@
+	flags.StringVar(&c.ttl, "ttl", "", "Time the variable will be locked")
+	flags.StringVar(&c.lockDelay, "delay", "", "Time the variable is blocked from locking when a lease is not renewed")
+	flags.BoolVar(&c.shell, "shell", true, "Use a shell to run the command (can set a custom shell via the SHELL "+"environment variable).")
+
+	if fileInfo, _ := os.Stdout.Stat(); (fileInfo.Mode() & os.ModeCharDevice) != 0 {
+		flags.StringVar(&c.varPutCommand.outFmt, "out", "none", "")
+	} else {
+		flags.StringVar(&c.varPutCommand.outFmt, "out", "json", "")
+	}
+
+	if err := flags.Parse(args); err != nil {
+		c.varPutCommand.Ui.Error(commandErrorText(c))
+		return 1
+	}
+
+	args = flags.Args()
+
+	// Manage verbose output
+	verbose := func(_ string) {} //no-op
+	if doVerbose {
+		verbose = func(msg string) {
+			c.varPutCommand.Ui.Warn(msg)
+		}
+	}
+
+	c.varPutCommand.verbose = verbose
+
+	if c.varPutCommand.Meta.namespace == "*" {
+		c.varPutCommand.Ui.Error(errWildcardNamespaceNotAllowed)
+		return 1
+	}
+
+	if len(args) < 2 {
+		c.varPutCommand.Ui.Error(fmt.Sprintf("Not enough arguments (expected >2, got %d)", len(args)))
+		return 1
+	}
+
+	path, args, err = c.readPathFromArgs(args)
+	if err != nil {
+		c.varPutCommand.Ui.Error(err.Error())
+		return 1
+	}
+
+	sv, err := c.varPutCommand.makeVariable(path)
+	if err != nil {
+		c.varPutCommand.Ui.Error(fmt.Sprintf("Failed to parse variable data: %s", err))
+		return 1
+	}
+
+	if sv.Lock == nil {
+		if c.ttl == "" && c.lockDelay == "" {
+			c.varPutCommand.verbose("Using defaults for the lock")
+		}
+
+		sv.Lock = &api.VariableLock{
+			TTL:       api.DefaultLockTTL.String(),
+			LockDelay: api.DefaultLockDelay.String(),
+		}
+	}
+
+	if c.ttl != "" {
+		c.varPutCommand.verbose("Using TTL for the lock of " + c.ttl)
+		_, err := time.ParseDuration(c.ttl)
+		if err != nil {
+			c.varPutCommand.Ui.Error(fmt.Sprintf("Invalid TTL: %s", err))
+			return 1
+		}
+
+		sv.Lock.TTL = c.ttl
+	}
+
+	if c.lockDelay != "" {
+		c.varPutCommand.verbose("Using delay for the lock of " + c.lockDelay)
+		_, err := time.ParseDuration(c.ttl)
+		if err != nil {
+			c.varPutCommand.Ui.Error(fmt.Sprintf("Invalid Lock Delay: %s", err))
+			return 1
+		}
+
+		sv.Lock.LockDelay = c.lockDelay
+	}
+
+	// Get the HTTP client
+	client, err := c.varPutCommand.Meta.Client()
+	if err != nil {
+		c.varPutCommand.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	// Set up the locks handler
+	l, err := client.Locks(api.WriteOptions{}, *sv)
+	if err != nil {
+		c.varPutCommand.Ui.Error(fmt.Sprintf("Error initializing lock handler: %s", err))
+		return 1
+	}
+
+	ctx := context.Background()
+
+	// Set up the lease handler
+	ll := client.NewLockLeaser(l)
+	c.varPutCommand.verbose("Attempting to acquire lock")
+
+	// Run the shell inside the protected function.
+	if err := ll.Start(ctx, func(ctx context.Context) error {
+		c.varPutCommand.verbose(fmt.Sprintf("Variable locked, ready to execute: %s", args[0]))
```

Comment:
Seeing as this is verbose logging, is there any harm in outputting the full command being executed?

I was a surprised my testing didn't do that:
```
$ ./pkg/darwin_arm64/nomad var lock -verbose /lock/jrasell nomad agent -dev
Writing to path "lock/jrasell"
Using defaults for the lock
Attempting to acquire lock
Variable locked, ready to execute: nomad
```

Reply:
Not at all, let's put it there